java/compaction-core/src/test/java/sleeper/compaction/job/CompactionJobRunTest.java

```diff
@@ -0,0 +1,193 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CompactionJobRunTest {
+    private static final String DEFAULT_TASK_ID_1 = "task-id-1";
+    private static final String DEFAULT_TASK_ID_2 = "task-id-2";
+
+    @Test
+    public void shouldReportNoJobRunsIfJobNotStarted() {
+        // Given
+        CompactionJobCreatedStatus created1 = CompactionJobCreatedStatus.builder()
+                .updateTime(Instant.parse("2022-09-23T09:23:00.012Z"))
+                .partitionId("partition1").childPartitionIds(null)
+                .inputFilesCount(11)
+                .build();
+        // When
+        CompactionJobStatus status = new CompactionJobStatusesBuilder()
+                .jobCreated("job1", created1)
+                .build().get(0);
+
+        // Then
+        assertThat(status.getJobRuns())
+                .isEmpty();
+    }
+
+    @Test
+    public void shouldRetrieveNoFinishedStatusIfNoMatchingStartedStatus() {
+        // Given
+        CompactionJobCreatedStatus created1 = CompactionJobCreatedStatus.builder()
+                .updateTime(Instant.parse("2022-09-23T09:23:00.012Z"))
+                .partitionId("partition1").childPartitionIds(null)
+                .inputFilesCount(11)
+                .build();
+        CompactionJobStartedStatus started1 = CompactionJobStartedStatus.updateAndStartTime(
+                Instant.parse("2022-09-23T09:23:30.012Z"),
+                Instant.parse("2022-09-23T09:23:30.001Z"));
+
+        // When
+        CompactionJobStatus status = new CompactionJobStatusesBuilder()
+                .jobCreated("job1", created1)
+                .jobStarted("job1", started1, DEFAULT_TASK_ID_1)
+                .build().get(0);
+
+        // Then
+        CompactionJobRun jobRun1 = status.getLatestJobRun();
+        assertThat(jobRun1.getStartedStatus())
+                .isEqualTo(started1);
+        assertThat(jobRun1.getFinishedStatus())
+                .isNull();
+    }
+
+    @Test
+    public void shouldRetrieveCorrectFinishedStatusMatchingStartedStatus() {
+        // Given
+        CompactionJobCreatedStatus created1 = CompactionJobCreatedStatus.builder()
+                .updateTime(Instant.parse("2022-09-23T09:23:00.012Z"))
+                .partitionId("partition1").childPartitionIds(null)
+                .inputFilesCount(11)
+                .build();
+        CompactionJobStartedStatus started2 = CompactionJobStartedStatus.updateAndStartTime(
+                Instant.parse("2022-09-24T09:23:30.012Z"),
+                Instant.parse("2022-09-24T09:23:30.001Z"));
+        CompactionJobFinishedStatus finished2 = CompactionJobFinishedStatus.updateTimeAndSummary(
+                Instant.parse("2022-09-24T09:24:00.012Z"),
+                new CompactionJobSummary(new CompactionJobRecordsProcessed(450L, 300L),
+                        Instant.parse("2022-09-24T09:23:30.001Z"),
+                        Instant.parse("2022-09-24T09:24:00.001Z")));
```

Comment:
It's a bit misleading that the start & finish statuses have a different number against them from the created one.

It might be clearer to remove the numbers from the variable names when there's only one.